fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/api/event/server/PlayerJoinCallback.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.server;
+
+import net.minecraft.network.ClientConnection;
+import net.minecraft.server.network.ServerPlayerEntity;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * This event is called when a player joins the server.
+ *
+ * <p>It hooks in at the end of {@link net.minecraft.server.PlayerManager#onPlayerConnect(ClientConnection, ServerPlayerEntity)} through {@link net.fabricmc.fabric.mixin.event.lifecycle.MixinPlayerManager}.
```

Comment:
Mentioning the impl really doesn't help with this, nor does mentioning the method it occurs in.

I'd say mention more concretely such as, "This event is called after the joining player has been put into the ServerWorld" or something like that

Reply:
> 
> 
> > giving items to the player when they first join
> 
> This seems to be quite a tall order since I don't see any logic to detect when a player has first joined the server. I know there is a compound tag you can check in one of the player manager calls as null to see if they are a new player

Isn't this one of those things that should be suggested after the initial PR is accepted? From what i understand, PRs for lacking functionality shouldn't need to be perfect from the start.